dashboard/src/components/Summary/Summary.tsx

```diff
@@ -0,0 +1,79 @@
+import { useMemo } from 'react';
+
+import BaseCard from '../Cards/BaseCard';
+import BaseTable from '../Table/BaseTable';
+import { TableBody, TableCell, TableRow } from '../ui/table';
+import ListingComponentItem, {
+  IListingComponentItem,
+} from '../ListingComponentItem/ListingComponentItem';
+
+export interface ISummary {
+  title: string;
+  summaryHeaders: string[];
+  summaryBody: ISummaryItem[];
+}
+
+export interface ISummaryItem {
+  arch: IListingComponentItem;
+  compilers: string[];
+}
+
+const Summary = ({
+  title,
+  summaryHeaders,
+  summaryBody,
+}: ISummary): JSX.Element => {
+  const summaryHeadersRow = useMemo(() => {
+    return summaryHeaders.map(header => <span key={header}>{header}</span>);
+  }, [summaryHeaders]);
+
+  const summaryBodyRows = useMemo(() => {
+    return summaryBody.map(row => (
+      <SummaryItem
+        key={row.arch.text}
+        arch={row.arch}
+        compilers={row.compilers}
+      />
+    ));
+  }, [summaryBody]);
```

Comment:
Nit: I feel like removing the unnecessary `return` will make everything cleaner, but it's up to you.

```suggestion
  const summaryBodyRows = useMemo(
    () =>
      summaryBody.map(row => (
        <SummaryItem
          key={row.arch.text}
          arch={row.arch}
          compilers={row.compilers}
        />
      )),
    [summaryBody],
  );
```